### ISSUE TITLE
trainer: _param_filter, filters trainable parameters by mistake.

### ISSUE THREAD

**Issue by XueBaolu** [alibaba/FederatedScope#733]
when _self.cfg.personalization.local_param = [], the function __param_filter_ will filter all of the trainable parameters. 
Perhaps it can be fixed by adding "if len(filter_keywords) == 0: return state_dict" in around 392 lines.

**Comment by yxdyc** [comment]
Thank you for your interest in FS!. When `filter_keywords=self.cfg.personalization.local_param = []`,  in line 396 of [trainer.py](https://github.com/alibaba/FederatedScope/blob/480b67de851df2fa02b7cc2189f1803d947998d7/federatedscope/core/trainers/trainer.py#L396), the function [filter_by_specified_keywords](https://github.com/alibaba/FederatedScope/blob/master/federatedscope/core/trainers/utils.py#L21) actually will skip its for loop so that `preserve` is always `True`. Thus the problem should not be caused by `personalization.local_param = []`.

Maybe your reported issue has been fixed in this [PR](https://github.com/alibaba/FederatedScope/pull/642). What are your specific test scripts and environment (especially the version of FS)?

**Comment by XueBaolu** [comment]
Thank you for responding. 
I download the codes recently, so the version of FS used is new. But when I run FL with resnet18 and CIFAR10, the accuracy of global model is 0.1 after the 100 rounds. And I found that the function _param_filter will filter all of the trainable parameters, because the code in 'filter(lambda elem ...' is always true. 
After adding "if len(filter_keywords) == 0: return state_dict" in 392 lines, the accuracy of the system becomes 0.5, a normal value.

**Comment by rayrayraykk** [comment]
Thank you for reaching out with your issue. We have attempted to reproduce the problem based on your description, but everything appears to be functioning as expected on our end:
`python federatedscope/main.py --cfg scripts/example_configs/femnist.yaml`


![image](https://github.com/alibaba/FederatedScope/assets/39145382/7ebfbeb4-9cb4-47d2-8063-d139a91441ee)

Additionally, here are the relevant git logs:
![image](https://github.com/alibaba/FederatedScope/assets/39145382/c4b67d22-919d-47cf-b4ae-5a5b3ccae5b3)

It's possible that the issue might be related to the model's trainability or an inadvertent switch to eval mode. Could you please double-check your model configuration and ensure it is set to training mode? If the issue persists, feel free to share more details or any error messages you encounter, and we'll be happy to take another look.

Thank you for using our project, and we are committed to helping you resolve this matter.

**Comment by XueBaolu** [comment]
Thank you, maybe something wrong, resulting in discrepancies in our experimental results. 
But it's pleased that this error has been rectified.
